packages/types/src/resource.ts

```diff
@@ -3,7 +3,7 @@ export type ClerkResourceReloadParams = {
 };
 
 export interface ClerkResource {
-  readonly id?: string;
+  readonly id?: string | undefined;
```

Comment:
:upside_down_face: If the `undefined` part was added to accomodate the `UserSettingsResource`, I would expect that simply making it optional would suffice :/ But I guess you checked it and the linter complains somewhere?

Reply:
Exactly! 

This fixes an issue where a Clerk package that has a direct dependency on `@clerk/types` (like `backend-core`) is installed within a TS application that includes the following options in its `tsconfig.json` file:

```
{
  "compilerOptions": {
    ...
    "exactOptionalPropertyTypes": true,
    "skipLibCheck": false,
  },
}
```

Having `skipLibCheck` set to false would make `tsc` type check the app and all the imported libs, and without the explicit `| undefined;` in our type, the `exactOptionalPropertyTypes` would cause an error.

More info: https://www.typescriptlang.org/tsconfig#exactOptionalPropertyTypes

Reply:
Thanks for the explanation!